### PR TITLE
164 jenny emb class

### PR DESCRIPTION
**Pull request by JennyFerriol**
* added embedded class component
* added drop down feature

### CODE REVIEW

File: src/frontend/src/Componenets/Class.tsx [hunk 1]
```diff
@@ -1,47 +1,80 @@
...
+                        <b>{data.id + "\n"}</b><br></br>
+                        {data.displayName}<br></br>
+                        <hr></hr>
+                        {data.desc}
+                        <hr></hr>
+                        {data.addl}
+                    </Tooltip>
+                    {isEmbeddedClassOpen && (
+                        <div className="embedded-class">
+                            <EmbeddedClass data={mockData}/>
```
Comment (ARES-79) [inline src/frontend/src/Componenets/Class.tsx:72]: Currently, the embedded class shows up in front of the Info Panel when hovering on a class. Can we get it to be behind the info panel?

File: src/frontend/src/Componenets/Class.css [hunk 2]
```diff
@@ -11,16 +16,46 @@
 .courseName{
     display:flex;
     justify-content: center;
-
     font-size: 12px;
+    text-align: center;
+    margin-bottom: 10px;
 }
-
 .class .classInfo  {
```
Comment (archie1802) [inline src/frontend/src/Componenets/Class.css:23]: we shouldnt be using px for size, gotta use percents

File: src/frontend/src/Componenets/Class.css [hunk 3]
```diff
@@ -11,16 +16,46 @@
...
     background: #a69355;
-    max-width: 25%;
+    max-width: 200px;
     opacity: 100;
+    z-index: 9999;
+}
+.collapsible {
+    position: absolute;
+    cursor: pointer;
+    padding: 10px;
```
Comment (archie1802) [inline src/frontend/src/Componenets/Class.css:32]: Same for all these values, should be percents

File: src/frontend/src/Componenets/Class.css [hunk 4]
```diff
@@ -11,16 +16,46 @@
...
+.embedded-class {
+    position: absolute;
+    top: 40%;
+    background: #FFF5D0;
+    margin: 8px;
+    border-radius: 5px;
+    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
+}
+.class.expanded {
+    height: 250px;
```
Comment (archie1802) [inline src/frontend/src/Componenets/Class.css:50]: Same here

File: src/frontend/src/Componenets/Class.tsx [hunk 5]
```diff
@@ -1,47 +1,80 @@
...
+}
 
 interface classProps {
     data: QuarterClassData
     index: number;
 }
+
+const mockData: EmbeddedQuarterClassData = {
+    id: "CSC 101",
+    displayName: "Introduction to Computer Science",
```
Comment (archie1802) [inline src/frontend/src/Componenets/Class.tsx:26]: Should this name be embeddedSemester class bc technically its a semester class

File: src/frontend/src/Componenets/EmbeddedClass.tsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import "./Class.css";
+import {Tooltip} from 'react-tooltip'
+import { Simulate } from 'react-dom/test-utils';
+import click = Simulate.click;
+
+export interface EmbeddedQuarterClassData {
```
Comment (archie1802) [inline src/frontend/src/Componenets/EmbeddedClass.tsx:7]: all interfaces should be in the interfaces folder

File: src/frontend/src/Componenets/Class.tsx (hunk 1 above)
Comment (JennyFerriol) [reply src/frontend/src/Componenets/Class.tsx:72]: done

File: src/frontend/src/Componenets/Class.css (hunk 2 above)
Comment (JennyFerriol) [reply src/frontend/src/Componenets/Class.css:23]: ok

File: src/frontend/src/Componenets/Class.css (hunk 3 above)
Comment (JennyFerriol) [reply src/frontend/src/Componenets/Class.css:32]: ok

File: src/frontend/src/Componenets/Class.css (hunk 4 above)
Comment (JennyFerriol) [reply src/frontend/src/Componenets/Class.css:50]: ok

File: src/frontend/src/Componenets/Class.tsx (hunk 5 above)
Comment (JennyFerriol) [reply src/frontend/src/Componenets/Class.tsx:26]: true

File: src/frontend/src/Componenets/EmbeddedClass.tsx (hunk 6 above)
Comment (JennyFerriol) [reply src/frontend/src/Componenets/EmbeddedClass.tsx:7]: ok

**Review by ARES-79 — APPROVED** [review]

**Pull request merged**